Detectors/EMCAL/simulation/include/EMCALSimulation/Patches.h

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file Patches.h
+/// \brief EMCAL Patches for the LZEROElectronics
+#ifndef ALICEO2_EMCAL_PATCH_H
+#define ALICEO2_EMCAL_PATCH_H
+
+#include <array>
+#include <vector>
+#include <gsl/span>
+#include "Rtypes.h"
+// #include "DataFormatsEMCAL/Cluster.h"
+// #include "DataFormatsEMCAL/Digit.h"
+// #include "DataFormatsEMCAL/Cell.h"
+// #include "EMCALBase/Geometry.h"
+
+// #include <fairlogger/Logger.h> // for LOG
+
+namespace o2
+{
+
+namespace emcal
+{
+
+/// \param mADCvalues vector of the ADC values
+/// \param previousTimebinADCvalue previous ADC value, popped out (i-4) if i refers to the current Timebin
+struct FastOrStruct {
+  std::vector<double> mADCvalues;
+  double mPreviousTimebinADCvalue = 0.;
+
+  /// Compute current timesum
+  double timesum()
+  {
+    double timesumvalue = 0;
+    for (auto ADCvalue : mADCvalues) {
+      timesumvalue += ADCvalue;
+    }
+    return timesumvalue;
+  }
+
+  void init()
+  {
+  }
+
+  // Update internal ADC values (4 timebins)
+  void updateADC(double ADCvalue)
+  {
+
+    if (mADCvalues.size() == 4) {
+      mPreviousTimebinADCvalue = mADCvalues.front();
+      mADCvalues.erase(mADCvalues.begin());
+    }
+    mADCvalues.push_back(ADCvalue);
+  }
+};
+
+/// \struct Patches
+/// \brief Patches creator, based on the Patches
+/// \ingroup EMCALsimulation
+/// \author Markus Fasel, ORNL
+/// \author Simone Ragoni, Creighton U.
+/// \date 03/12/2022
+///
+
+struct Patches {
```

Comment:
As mentioned in the cxx file it is essentially the simulation of a TRU, so call it `TRUElectronics`

Reply:
Done